diff --git a/src/oxygenthemingengine.cpp b/src/oxygenthemingengine.cpp
index be377af..60f33a5 100644
--- a/src/oxygenthemingengine.cpp
+++ b/src/oxygenthemingengine.cpp
@@ -1025,23 +1025,27 @@ namespace Oxygen
             if( Gtk::gtk_widget_path_has_type( path, GTK_TYPE_FRAME ) )
             { Style::instance().renderGroupBoxBackground( context, widget, x, y, w, h, Blend ); }

-            MenuBarStateEngine& engine( Style::instance().animations().menuBarStateEngine() );
-            engine.registerWidget(widget);
-
-            // draw animated or fade-out rect
-            if( engine.animatedRectangleIsValid( widget ) )
+            if( widget )
             {
+                MenuBarStateEngine& engine( Style::instance().animations().menuBarStateEngine() );
+                engine.registerWidget(widget);

-                const GdkRectangle& rect( engine.animatedRectangle( widget ) );
-                Style::instance().renderMenuItemRect( context, 0L, engine.widget( widget, AnimationCurrent ), rect.x, rect.y, rect.width, rect.height, Hover|Blend );
+                // draw animated or fade-out rect
+                if( engine.animatedRectangleIsValid( widget ) )
+                {

-            } else if( engine.isAnimated( widget, AnimationPrevious ) ) {
+                    const GdkRectangle& rect( engine.animatedRectangle( widget ) );
+                    Style::instance().renderMenuItemRect( context, 0L, engine.widget( widget, AnimationCurrent ), rect.x, rect.y, rect.width, rect.height, Hover|Blend );

-                const AnimationData data( engine.animationData( widget, AnimationPrevious ) );
-                const GdkRectangle& rect( engine.rectangle( widget, AnimationPrevious ) );
-                Style::instance().renderMenuItemRect( context, 0L, engine.widget( widget, AnimationPrevious ), rect.x, rect.y, rect.width, rect.height, Hover|Blend, data );
+                } else if( engine.isAnimated( widget, AnimationPrevious ) ) {

+                    const AnimationData data( engine.animationData( widget, AnimationPrevious ) );
+                    const GdkRectangle& rect( engine.rectangle( widget, AnimationPrevious ) );
+                    Style::instance().renderMenuItemRect( context, 0L, engine.widget( widget, AnimationPrevious ), rect.x, rect.y, rect.width, rect.height, Hover|Blend, data );
+
+                }
             }
+
             return;

         } else if( gtk_widget_path_is_type( path, GTK_TYPE_MENU ) ) {
